client/app.jsx

```diff
@@ -1,8 +1,61 @@
 import React from 'react';
 import Home from './pages/home';
+import parseRoute from './lib/parse-route';
+import UserPost from './components/post';
+import CreatePost from './pages/create-post';
 
 export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      route: parseRoute(window.location.hash),
+      post: []
+    };
+    this.getPosts = this.getPosts.bind(this);
+  }
+
+  componentDidMount() {
+    window.addEventListener('hashchange', event => {
+      const route = parseRoute(window.location.hash);
+      this.setState({
+        route
+      });
+    });
+    fetch('/api/posts')
+      .then(res => res.json())
+      .then(post => this.setState({
+        post
+      }));
```

Comment:
Here you should just call your `getPosts` method instead of rewriting out the same `fetch` request